ShelterModule/Services/Implementations/Announcements/AnnouncementCommand.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Database;
+using Microsoft.EntityFrameworkCore;
+using ShelterModule.Models.Announcements;
+using ShelterModule.Services.Interfaces.Announcements;
+
+namespace ShelterModule.Services.Implementations.Announcements;
+
+public class AnnouncementCommand : IAnnouncementCommand
+{
+    private readonly PetShareDbContext _dbContext;
+
+    public AnnouncementCommand(PetShareDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    public async Task<Announcement> AddAsync(Announcement announcement, CancellationToken token = default)
+    {
+        var entityAnnouncement = announcement.ToEntity();
+        _dbContext.Add(entityAnnouncement);
+        await _dbContext.SaveChangesAsync(token);
+        return announcement;
+    }
+
+    public async Task<Announcement?> UpdateAsync(Guid id, AnnouncementPutRequest request,
+        CancellationToken token = default)
+    {
+        var entityToUpdate = await _dbContext.Announcements.Include(x => x.Pet).
+                                              Include(x => x.Author).
```

Comment:
We don't actually need to `Include()` anything there. We can modify linked IDs without includes. They are only used to load all the data of linked entities, which we don't need here.